### ISSUE TITLE
Gettin xml.etree.ElementTree.ParseError

### ISSUE THREAD

**Issue by asdfasdfasdfasdfasasdf** [zeroepoch/plotbitrate#29]
I get the same error no matter if I use the pip installer or if I use the plotbitrate.py directly:

python3 plotbitrate.py -f xml_raw -o frames.xml file.mp4
Traceback (most recent call last):
  File "/home/.../plotbitrate.py", line 691, in <module>
    main()
  File "/home/.../plotbitrate.py", line 637, in main
    save_raw_xml(
  File "/home/.../plotbitrate.py", line 274, in save_raw_xml
    try_get_frame_time_from_node(eTree.fromstring(line))
  File "/usr/lib/python3.10/xml/etree/ElementTree.py", line 1343, in XML
    return parser.close()
xml.etree.ElementTree.ParseError: no element found: line 2, column 0

Using Python 3.10.12 in Kubuntu 22.04.

"python3 plotbitrate.py file.mp4" and "python3 plotbitrate.py -f csv_raw -o frames.csv file.mp4" works fine.

**Comment by zeroepoch** [comment]
Hi @asdfasdfasdfasdfasasdf. Not sure exactly what is wrong without seeing the XML output from `ffprobe`. You can try running the command with `--no-progress`, which will avoid this part of the code. Maybe once you have the XML saved you can check if see if anything looks unusual in the file. It's trying to parse the `<frame...>` lines to estimate the progress.

**Comment by asdfasdfasdfasdfasasdf** [comment]
Hi.
Thanks for the suggestion. Using `--no-progress` works. The resulting xml seems fine, I looked at it using `python3 plotbitrate.py frames.xml`.

Is there any way I can help you debug the progress part that isn't working?

**Comment by zeroepoch** [comment]
You could try using the following patch to print the line before parsing it for the time to see what it looks like.

```
diff --git a/plotbitrate.py b/plotbitrate.py
index 17e0683..da97516 100755
--- a/plotbitrate.py
+++ b/plotbitrate.py
@@ -270,6 +270,7 @@ def save_raw_xml(
                 if not no_progress \
                         and duration > 0 \
                         and line.lstrip().startswith(b"<frame "):
+                    print(line)
                     frame_time = \
                         try_get_frame_time_from_node(eTree.fromstring(line))
```

Looking at the code it does write to the file first before it attempts to print the progress, so you might be able to just look at the last line of the file when it fails to see what it looks like. This is assuming the file contents get flushed to disk before it exits.

**Comment by asdfasdfasdfasdfasasdf** [comment]
I get one line extra (the first one below) if I add that print:
`
b'        <frame pkt_pts_time="0.000000" best_effort_timestamp_time="0.000000" pkt_size="901" pict_type="I">\n'
Traceback (most recent call last):
  File "/home/.../plotbitrate.py", line 692, in <module>
    main()
  File "/home/.../plotbitrate.py", line 638, in main
    save_raw_xml(
  File "/home/.../plotbitrate.py", line 275, in save_raw_xml
    try_get_frame_time_from_node(eTree.fromstring(line))
  File "/usr/lib/python3.10/xml/etree/ElementTree.py", line 1343, in XML
    return parser.close()
xml.etree.ElementTree.ParseError: no element found: line 2, column 0
`

**Comment by zeroepoch** [comment]
I see the same error locally when using this string you provided.

```
>>> import xml.etree.ElementTree
>>> xml.etree.ElementTree.fromstring(b'        <frame pkt_pts_time="0.000000" best_effort_timestamp_time="0.000000" pkt_size="901" pict_type="I">\n')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python3.11/xml/etree/ElementTree.py", line 1339, in XML
    return parser.close()
           ^^^^^^^^^^^^^^
xml.etree.ElementTree.ParseError: no element found: line 2, column 0
```

I think it's because the XML tag isn't closed. If I close the frame tag with `/>` then it parses fine. I guess that means the frame line is not just a single line anymore. Would it be possible to paste the first few lines of the XML file produced when the progress indicator is disabled?

**Comment by asdfasdfasdfasdfasasdf** [comment]
`<?xml version="1.0" encoding="UTF-8"?>
<ffprobe>
    <frames>
        <frame pkt_pts_time="0.000000" best_effort_timestamp_time="0.000000" pkt_size="899" pict_type="I">
            <side_data_list>
                <side_data />
            </side_data_list>
        </frame>
        <frame pkt_pts_time="0.040000" best_effort_timestamp_time="0.040000" pkt_size="21" pict_type="B"/>
        <frame pkt_pts_time="0.080000" best_effort_timestamp_time="0.080000" pkt_size="21" pict_type="B"/>
        <frame pkt_pts_time="0.120000" best_effort_timestamp_time="0.120000" pkt_size="22" pict_type="B"/>
        <frame pkt_pts_time="0.160000" best_effort_timestamp_time="0.160000" pkt_size="21" pict_type="B"/>
        <frame pkt_pts_time="0.200000" best_effort_timestamp_time="0.200000" pkt_size="21" pict_type="B"/>
        <frame pkt_pts_time="0.240000" best_effort_timestamp_time="0.240000" pkt_size="25" pict_type="P"/>
        <frame pkt_pts_time="0.280000" best_effort_timestamp_time="0.280000" pkt_size="21" pict_type="B"/>
        <frame pkt_pts_time="0.320000" best_effort_timestamp_time="0.320000" pkt_size="21" pict_type="B"/>
        <frame pkt_pts_time="0.360000" best_effort_timestamp_time="0.360000" pkt_size="24" pict_type="B"/>
        <frame pkt_pts_time="0.400000" best_effort_timestamp_time="0.400000" pkt_size="21" pict_type="B"/>
        <frame pkt_pts_time="0.440000" best_effort_timestamp_time="0.440000" pkt_size="21" pict_type="B"/>
        <frame pkt_pts_time="0.480000" best_effort_timestamp_time="0.480000" pkt_size="28" pict_type="P"/>
        <frame pkt_pts_time="0.520000" best_effort_timestamp_time="0.520000" pkt_size="21" pict_type="B"/>
        <frame pkt_pts_time="0.560000" best_effort_timestamp_time="0.560000" pkt_size="21" pict_type="B"/>
        <frame pkt_pts_time="0.600000" best_effort_timestamp_time="0.600000" pkt_size="24" pict_type="B"/>
        <frame pkt_pts_time="0.640000" best_effort_timestamp_time="0.640000" pkt_size="21" pict_type="B"/>
        <frame pkt_pts_time="0.680000" best_effort_timestamp_time="0.680000" pkt_size="21" pict_type="B"/>
        <frame pkt_pts_time="0.720000" best_effort_timestamp_time="0.720000" pkt_size="28" pict_type="P"/>
        <frame pkt_pts_time="0.760000" best_effort_timestamp_time="0.760000" pkt_size="21" pict_type="B"/>
        <frame pkt_pts_time="0.800000" best_effort_timestamp_time="0.800000" pkt_size="21" pict_type="B"/>
        <frame pkt_pts_time="0.840000" best_effort_timestamp_time="0.840000" pkt_size="24" pict_type="B"/>
        <frame pkt_pts_time="0.880000" best_effort_timestamp_time="0.880000" pkt_size="21" pict_type="B"/>`

**Comment by asdfasdfasdfasdfasasdf** [comment]
Here is the entire file:
[frames.tar.gz](https://github.com/zeroepoch/plotbitrate/files/13302228/frames.tar.gz)

**Comment by zeroepoch** [comment]
Interesting... The first frame of the file has some extra data, which appears to be mostly empty, but nevertheless still there.

```
        <frame pkt_pts_time="0.000000" best_effort_timestamp_time="0.000000" pkt_size="899" pict_type="I">
            <side_data_list>
                <side_data />
            </side_data_list>
        </frame>
```

Which version of ffprobe do you have (`ffprobe --version`)? Not sure if this is something specific to the file you're using or the FFMpeg version. I'm using 6.0, which is the latest version from what I can see and on a few files I usually try it's not producing this extra `side_data` tag. Do you only see it on a specific file?

**Comment by zeroepoch** [comment]
Oh... I have a file which is also doing it now, so maybe something changed with recent versions of FFMpeg. At least I can reproduce the issue so that will help me to debug it. I'll see if I can come up with a workaround this weekend. Worst case I can just ignore the parse error and skip the frame since it's just 1 frame and shouldn't skew the progress.

**Comment by zeroepoch** [comment]
@asdfasdfasdfasdfasasdf, I just released version 1.1.1.1 which should workaround this issue. I couldn't find a good way to omit the side_data tags with ffprobe, so I just ignore them essentially by closing the opening tag myself. We only need the frame tag attributes to get the current time for the progress. Ignoring "broken" lines wouldn't have worked very well since I found a recent movie I encoded had this on every frame because it was a 4k HDR file. If for some reason patching the line doesn't work it will at least ignore the error now and continue rather than crash. Give it a try and let me know if it resolves the issue.
